### PR TITLE
Use consts defined in api instead of defining another ones.

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
empty_dir defines some consts. There are already similar consts in api types. So remove the local ones in empty_dir.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sjenning** [comment]
Only concern is maybe adding `StorageMediumUnknown` to the API, since that is kinda an empty_dir internally thing i.e. one can not create a volume of type `Unknown`

Otherwise I like the cleanup :+1:

**Comment by xiangpengzhao** [comment]
@sjenning thanks! IMO,  adding `StorageMediumUnknown` to the API is not expecting users to create such a volume. It's just a placeholder for all invalid storage media.

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -944,8 +944,9 @@ type StorageMedium string
 
 const (
 	StorageMediumDefault   StorageMedium = ""          // use whatever the default is for the node
-	StorageMediumMemory    StorageMedium = "Memory"    // use memory (tmpfs)
+	StorageMediumMemory    StorageMedium = "Memory"    // use memory (e.g. tmpfs on linux)
 	StorageMediumHugepages StorageMedium = "HugePages" // use hugepages
+	StorageMediumUnknown   StorageMedium = "Unknown"   // assume anything we don't explicitly handle is this
```
Comment (derekwaynecarr) [inline staging/src/k8s.io/api/core/v1/types.go]: this is awkward as a separate pr will have this fail validation.

**Review by derekwaynecarr — COMMENTED** [review]
i like having a single list of consts, i just find the unknown bit awkward.

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (xiangpengzhao) [reply staging/src/k8s.io/api/core/v1/types.go]: Do you mean PR #52936? It just validate by `validStorageMedia` which doesn't include `StorageMediumUnknown`.

**Comment by xiangpengzhao** [comment]
>i like having a single list of consts

@derekwaynecarr I'd like to get more suggestions on this (please see my inline reply). Thanks!

**Comment by derekwaynecarr** [comment]
i am going to defer to a member of the api review team to see if there are any issues or concerns.

it seems ok as long as validation doesn't allow it.

/assign @liggitt 
/unassign

**Comment by xiangpengzhao** [comment]
@derekwaynecarr @liggitt validation doesn't validate it but the emptyDir volumes does.

https://github.com/kubernetes/kubernetes/blob/8e8601a1cbf9400ea15116b1591c03e72dfcb0bb/pkg/apis/core/validation/validation.go#L438-L452

https://github.com/kubernetes/kubernetes/blob/dd9981d038012c120525c9e6df98b3beb3ef19e1/pkg/volume/empty_dir/empty_dir.go#L225-L234

It won't allow `StorageMediumUnknown` type (in default).

File: staging/src/k8s.io/api/core/v1/types.go [hunk 2]
```diff
@@ -1009,8 +1009,9 @@ type StorageMedium string
 
 const (
 	StorageMediumDefault   StorageMedium = ""          // use whatever the default is for the node
-	StorageMediumMemory    StorageMedium = "Memory"    // use memory (tmpfs)
+	StorageMediumMemory    StorageMedium = "Memory"    // use memory (e.g. tmpfs on linux)
 	StorageMediumHugePages StorageMedium = "HugePages" // use hugepages
+	StorageMediumUnknown   StorageMedium = "Unknown"   // assume anything we don't explicitly handle is this
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go]: agree that this looks too much like a legitimate value

File: pkg/volume/empty_dir/empty_dir.go [hunk 3]
```diff
@@ -155,20 +155,12 @@ func (plugin *emptyDirPlugin) ConstructVolumeSpec(volName, mountPath string) (*v
...
 	// on a memory FS (tmpfs on Linux) but is not the root mountpoint of
 	// that tmpfs.
-	GetMountMedium(path string) (storageMedium, bool, error)
+	GetMountMedium(path string) (v1.StorageMedium, bool, error)
 }
 
-type storageMedium int
-
-const (
-	mediumUnknown   storageMedium = 0 // assume anything we don't explicitly handle is this
```
Comment (liggitt) [inline pkg/volume/empty_dir/empty_dir.go]: does this not map reasonably to `v1.StorageMediumDefault`?

Comment (xiangpengzhao) [reply pkg/volume/empty_dir/empty_dir.go]: @liggitt :+1: sounds reasonable to me! I update the PR. PTAL. Thanks!

**Comment by xiangpengzhao** [comment]
@liggitt squashed. PTAL. Thanks!

**Comment by xiangpengzhao** [comment]
/assign @thockin @saad-ali
for additional approval. Thanks!

**Pull request merged**